### PR TITLE
p:d:GridRefinement: accept other types of triangulations

### PR DESCRIPTION
**Pull request by peterrum**
FYI @mschreter

### CODE REVIEW

**Review by marcfehling — COMMENTED** [review]
Can you add a `dynamic_cast` check for `parallel::DistributedTriangulationBase`?

**Review by drwells — APPROVED** [review]

**Comment by marcfehling** [comment]
> As far as I see it, the functions are working for any triangulation. So I don't want to add a check. Furthermore, I see this more as a first step to move make these functions part of the normal namespace.

Yes, they should work for any Triangulation. I use some of the functionality also in the `hp::Refinement` namespace.

**Comment by marcfehling** [comment]
```
/jenkins/workspace/dealii_PR-14072/source/distributed/grid_refinement.cc:108:44: error: ‘const class dealii::Triangulation<2, 3>’ has no member named ‘n_locally_owned_active_cells’
     Assert(locally_owned_indicators.size()
```

**Comment by marcfehling** [comment]
I assume you want to use this functionality in the context of `parallel::shared::Triangulation`? Let's wait on more feedback to https://github.com/dealii/dealii/pull/14073#issuecomment-1169237711 before merging this patch...

**Comment by peterrum** [comment]
> I assume you want to use this functionality in the context of parallel::shared::Triangulation? Let's wait on more feedback to https://github.com/dealii/dealii/pull/14073#issuecomment-1169237711 before merging this patch...

No that one is unrelated.

File: include/deal.II/distributed/grid_refinement.h
Comment (kronbichler) [inline include/deal.II/distributed/grid_refinement.h:156]: The initial comment of this function still says `Like dealii::GridRefinement::refine_and_coarsen_fixed_number, but for parallel distributed triangulations.` - I am wondering if we can use a simpler-to-recognize name, e.g. `Like dealii::GridRefinement::refine_and_coarsen_fixed_number, but designed for parallel computations.` Together with the rest of the description, this would seem to be more accurate.

**Review by kronbichler — APPROVED** [review]
Looks good otherwise.

**Review by kronbichler — APPROVED** [review]

**Review by masterleinad — APPROVED** [review]

**Pull request merged**
